src/Microsoft.AspNetCore.Blazor/Components/EventHandlers.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    [EventHandler("onchange", typeof(UIChangeEventArgs))]
+    [EventHandler("onclick", typeof(UIMouseEventArgs))]
```

Comment:
For now I just coded up the thing you had support for. Do you have a good idea of what events we should put in the box or do you think I should try to come up with that list.

Reply:
Are the ones here the only ones that will compile correctly, or for unrecognized event names, do we fall back on some default type? It would be ideal if we did have a default type (that supplies no info other than that the event happened) for arbitrary event names. I know there's also work on the JS side to make arbitrary events work but that's separate from the compile-time work.

For the actual list of event names, we could use this list as inspiration: https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L3475, which also gives hints about the possible event arg types. Assuming we have support for default fallback for unknown events, we could probably reduce our set of explicitly-typed ones down to the most common ones:

```
    "activate": UIEvent;
    "beforeactivate": UIEvent;
    "beforedeactivate": UIEvent;
    "blur": FocusEvent;
    "change": Event;
    "click": MouseEvent;
    "contextmenu": PointerEvent;
    "dblclick": MouseEvent;
    "deactivate": UIEvent;
    "drag": DragEvent;
    "dragend": DragEvent;
    "dragenter": DragEvent;
    "dragleave": DragEvent;
    "dragover": DragEvent;
    "dragstart": DragEvent;
    "drop": DragEvent;
    "focus": FocusEvent;
    "input": Event;
    "invalid": Event;
    "keydown": KeyboardEvent;
    "keypress": KeyboardEvent;
    "keyup": KeyboardEvent;
    "mousedown": MouseEvent;
    "mousemove": MouseEvent;
    "mouseout": MouseEvent;
    "mouseover": MouseEvent;
    "mouseup": MouseEvent;
    "mousewheel": WheelEvent;
    "pause": Event;
    "play": Event;
    "playing": Event;
    "progress": ProgressEvent;
    "readystatechange": Event;
    "scroll": UIEvent;
    "seeked": Event;
    "seeking": Event;
    "select": UIEvent;
    "selectionchange": Event;
    "selectstart": Event;
    "stop": Event;
    "submit": Event;
    "touchcancel": TouchEvent;
    "touchend": TouchEvent;
    "touchmove": TouchEvent;
    "touchstart": TouchEvent;
    "volumechange": Event;
```

I know you might not want to go declaring new `UIEventArgs` subtypes for all these (and dozens of E2E tests) during this PR, so I think it would be fine to regard "detailed support for loads of DOM event types" as a separate work item.